internal/pkg/term/progress/cloudformation.go

```diff
@@ -239,6 +265,46 @@ func (c *stackComponent) addResourceRenderer(ev stream.StackEvent, description s
 	c.resources = append(c.resources, ListeningResourceRenderer(c.stack, ev.LogicalResourceID, description, opts))
 }
 
+// stackSetComponent is a DynamicRenderer that can display stack set operation events.
+type stackSetComponent struct {
+	stream <-chan stream.StackSetOpEvent
+	done   chan struct{}
+
+	style     RenderOptions
+	title     string
+	separator rune
+
+	mu        sync.Mutex
+	statuses  []cfnStatus
+	stopWatch *stopWatch
+}
+
+// Listen consumes stack set operation events and updates the status until the streamer closes the channel.
+func (c *stackSetComponent) Listen() {
+	for ev := range c.stream {
+		updateComponentStatus(&c.mu, &c.statuses, cfnStatus{
+			value:  ev.Operation.Status,
+			reason: ev.Operation.Reason,
+		})
+		updateComponentTimer(&c.mu, c.statuses, c.stopWatch)
+	}
+	close(c.done)
+}
+
+// Render renders the stack set status updates to out.
+func (c *stackSetComponent) Render(out io.Writer) (numLines int, err error) {
```

Comment:
Maybe put that into the comment?
```suggestion
func (c *stackSetComponent) Render(out io.Writer) (int, error) {
```

Reply:
Done!